src/Integration.Vsix/Analysis/DisableRuleCommand.cs

```diff
@@ -191,11 +188,28 @@ private static string FindErrorCodeForEntry(ITableEntryHandle handle)
             SonarRuleRepoKeys.Cpp
         };
 
-        private static bool IsDisablingRulesSupported(string errorCode)
+        private static void CalculateStatuses(SonarCompositeRuleId rule, SonarLintMode mode, out bool isVisible, out bool isEnabled)
         {
+            // Special case: JavaScript
+            // We don't support connected mode for JavaScript at the moment so we allow disabling
+            // in standalone mode. See #770.
+            if (SonarRuleRepoKeys.AreEqual(SonarRuleRepoKeys.JavaScript, rule.RepoKey))
```

Comment:
Is this block really needed? There's no connected mode in JS, so it should always work in the following block.

Reply:
Connected mode is set at solution level, so it is possible to have JS files in connected mode.